lnwallet/wallet.go

```diff
@@ -892,10 +903,22 @@ func (l *LightningWallet) currentNumAnchorChans() (int, error) {
 	}
 
 	var numAnchors int
-	for _, c := range chans {
+	cntChannel := func(c *channeldb.OpenChannel) {
+		// We skip private channels, as we assume they won't be used
+		// for routing.
```

Comment:
I believe this assumption is incorrect because of non-strict forwarding. If one has a public channel and a private one with the same node, then it's possible to forward through both.

Disclaimer: I really don't understand anchor channels properly yet, so I may be wrong or this is non-issue.